graylog2-web-interface/src/views/components/sidebar/SideBar.jsx

```diff
@@ -1,18 +1,77 @@
 // @flow strict
 import * as React from 'react';
 import PropTypes from 'prop-types';
+import styled from 'styled-components';
 
-import { AddWidgetButton, SearchResultOverview } from 'views/components/sidebar';
+import { AddWidgetButton } from 'views/components/sidebar';
 import { Icon, Spinner } from 'components/common';
 import type { ViewMetaData } from 'views/stores/ViewMetadataStore';
 
-import styles from './SideBar.css';
 import CustomPropTypes from '../CustomPropTypes';
 import HighlightingRules from './highlighting/HighlightingRules';
-import NavItem from './NavItem';
-
-const defaultNewViewTitle = 'New View';
-const defaultNewViewSummary = 'No summary.';
+import NavItem, { Title as NavItemTitle } from './NavItem';
+import ViewDescription from './ViewDescription';
+
+const defaultNewViewTitle = 'New Search';
+
+const Container: React.ComponentType<{ open: boolean }> = styled.div`
+  grid-area: sidebar;
+  z-index: 3;
+  background: #393939;
+  color: #9e9e9e;
+  height: calc(100vh - 50px);
+  padding-top: 20px;
+  position: sticky;
+  top: 50px;
+  grid-column-start: 1;
+  grid-column-end: ${props => (props.open ? 3 : 2)};
+  box-shadow: 3px 0 3px rgba(0, 0, 0, .25);
+`;
+
+const ContentOverlay = styled.div`
+  position: fixed;
+  top: 0;
+  bottom: 0;
+  left: 300px;
+  right: 0;
+  background: rgba(3, 3, 3, 0.25);
+`;
+
+const SidebarHeader: React.ComponentType<{open: boolean, hasTitle: boolean}> = styled(NavItemTitle)`
+  ${({ open, hasTitle }) => {
+    let justifyContent = 'center';
+    if (open && hasTitle) justifyContent = 'space-between';
+    if (open && !hasTitle) justifyContent = 'flex-end';
+    return `justify-content: ${justifyContent}`;
+  }}
+`;
+
+const Headline = styled.h3`
+  color: inherit;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`;
+
+const ToggleIcon = styled.div`
+  width: 25px;
+  text-align: center;
+  font-size: 20px;
+  cursor: pointer;
+`;
+
+const HorizontalRuler = styled.div`
+  width: 100%;
+  padding: 0 10px;
+  margin: 5px 0 10px 0;
+  
+  &::after {
+    content: ' ';
+    display: block;
+    width: 100%;
+    border-bottom: 1px solid currentColor;
+  }
+`;
 
```

Comment:
This discussion is maybe something for the frontend chat:

What do you think about creating a separate file for these styled-components as well, when they are to many? In my opinion the component looks not that clear / readable with 60 lines of styling.

Reply:
I don't think there is much to discuss here. If a component grows too big, it is a good idea to move it to a separate file.

Reply:
Great! I've created a new file for the styles. We should just define / discuss about a convention for the file name.
I don't like the idea (anymore) to have a file like `./styles.jsx` inside each component directory.
So far, I've named the file `Sidebar.styles.jsx`.